dodo.c

```diff
@@ -868,15 +894,67 @@ int execute(struct Program *p){
 }
 
 
+/* frees the elements of the linked list of instructions
+ * allocated while parsing
+ */
+void scrub(struct Program *p)
+{
+    struct Instruction *now = 0;
+    struct Instruction *next = 0;
+    if ( p->start ){
+        now = p->start;
+        do {
+            next = now->next;
+            free(now);
+            now = next;
+        } while( next );
+    }
+    p->start = NULL;
+}
+
+int repl(struct Program *p){
+    int exit_code = EXIT_FAILURE;
+    char line[4096]; /* FIXME: Perhaps use slurp-like behaviour instead */
+
+    /* FIXME: doesn't handle quit command */
+    while( 1 ){
+        printf("dodo: ");
+        p->source = fgets(line, sizeof(line), stdin);
+
+        if ( ! p->source ){
+            if ( feof(stdin) ){
+                exit_code = EXIT_SUCCESS;
+            } else {
+                printf("fgets failed in repl\n");
+                exit_code = EXIT_FAILURE;
+            }
+            goto EXIT;
+        }
+
+        /* note we don't error-out on parse or execute,
+         * keep the repl rolling */
```

Comment:
I do want to make sure we (eventually) provide helpful feedback on error,
also we should probably check parse for errors and if any occurred then skip the next execute phase.